crates/rooch/src/commands/account/commands/update.rs

```
@@ -0,0 +1,144 @@
+                    scheme.to_owned()
+                );
+
+                let (address, module_name) = match scheme {
+                    BuiltinScheme::Ed25519 => (
+                        Ed25519Validator::MODULE_ADDRESS,
+                        Ed25519Validator::MODULE_NAME,
+                    ),
+                    BuiltinScheme::MultiEd25519 => todo!(),
+                    BuiltinScheme::Ecdsa => (
+                        EcdsaK1Validator::MODULE_ADDRESS,
+                        EcdsaK1Validator::MODULE_NAME,
+                    ),
+                    BuiltinScheme::EcdsaRecoverable => (
+                        EcdsaK1RecoverableValidator::MODULE_ADDRESS,
+                        EcdsaK1RecoverableValidator::MODULE_NAME,
+                    ),
+                    BuiltinScheme::Schnorr => (
+                        SchnorrValidator::MODULE_ADDRESS,
+                        SchnorrValidator::MODULE_NAME,
+                    ),
+                };
+
+                let validator_struct_arg: Box<StructTag> =
+                    scheme.create_validator_struct_tag(address, module_name.to_string())?;
+
+                let signer = bcs::to_bytes(&existing_address).unwrap();
+                let public_key_bytes_vec = public_key.as_ref().to_vec();
+
+                let rotate_authentication_key_entry_function = Lazy::new(|| {
+                    create_function_id(
+                        address,
+                        module_name.as_str(),
+                        "rotate_authentication_key_entry",
+                    )
+                })
+                .clone();
+                let action = MoveAction::new_function_call(
+                    rotate_authentication_key_entry_function,
+                    vec![TypeTag::Struct(validator_struct_arg)],
+                    vec![signer, public_key_bytes_vec],
+                );
+
+                context
```

Need to handle the result, if this transaction failed, should return the error.

---

I'll do.